### ISSUE TITLE
Returned predictions of GAT/GCN models differs from GraphSAGE/HinSAGE models

### ISSUE THREAD

**Issue by adocherty** [stellargraph/stellargraph#375]
**Description**
There are currently a few differences between the GCN/GAT model API and the GraphSAGE model API. A significant one is the predictions returned by the full-batch models are all nodes, whereas the predictions returned by the sampled models are only those nodes supplied in the generator.

Essentially, the GCN/GAT models ignore the nodes passed to the `FullBatchNodeGenerator.flow` method when making predictions. They are only used to mask the nodes in the loss function for training.

Thus, when making predictions regardless of what is passed to the `.flow` method predictions for all nodes are returned.

For example, using the GAT example notebook the predictions returned when 10 nodes are specified is the same as when all nodes are provided:
```
all_nodes = node_data.index
all_gen = generator.flow(all_nodes)
all_predictions = model.predict_generator(all_gen)

some_gen = generator.flow(all_nodes[:10])
some_predictions = model.predict_generator(some_gen)

all_predictions.shape
> (2708, 7)

np.all(some_predictions == all_predictions)
> True
```

This additionally causes the problem that class weights passed manually to the `.fit_generator` function are ignored as they are used internally to mask the nodes not used in training (as noted by @PantelisElinas ).

**Expected behavior**
The Keras `.predict_generator` function returns predictions for only the nodes provided to the `.flow` method.

Additionally, we would like to be able to pass weights in the training functions to correctly deal with weighted problems.

Note: the expected behaviour should be able to be implemented using the `K.gather` (https://keras.io/backend/) function in the GCN/GAT models.

### Done Checklist (Development)
- [x] Produced code for required functionality
- [x] Branch and Pull Request build on CI
- [x] Branch and Pull Request pass unit tests on CI
- [x] Version number reflects new status
- [x] Peer Code Review Performed
- [x] Documentation in repo
- [x] CHANGELOG.md updated
- [x] Team demo
